code/modules/mob/living/living.dm

```diff
@@ -774,26 +774,54 @@ default behaviour is:
 	handle_footstep(loc)
 
 	if(pulling) // we were pulling a thing and didn't lose it during our move.
+		var/pull_dir = get_dir(src, pulling)
+		
 		if(pulling.anchored || !isturf(pulling.loc))
 			stop_pulling()
-			return
-
-		var/pull_dir = get_dir(src, pulling)
-		if(get_dist(src, pulling) > 1 || (moving_diagonally != SECOND_DIAG_STEP && ((pull_dir - 1) & pull_dir))) // puller and pullee more than one tile away or in diagonal position
+			
+		else if(get_dist(src, pulling) > 1 || (moving_diagonally != SECOND_DIAG_STEP && ((pull_dir - 1) & pull_dir))) // puller and pullee more than one tile away or in diagonal position
 			// If it is too far away or across z-levels from old location, stop pulling.
 			if(get_dist(pulling.loc, oldloc) > 1 || pulling.loc.z != oldloc?.z)
 				stop_pulling()
-				return
 
 			// living might take damage from drags
-			if(isliving(pulling))
+			else if(isliving(pulling))
 				var/mob/living/M = pulling
 				M.dragged(src, oldloc)
 
 			pulling.Move(oldloc, 0, movetime) // the pullee tries to reach our previous position
 			if(pulling && get_dist(src, pulling) > 1) // the pullee couldn't keep up
 				stop_pulling()
 
+	if(!isturf(loc))
+		return	
+	else if(lastarea?.has_gravity == 0)
+		inertial_drift()
+	//VOREStation Edit Start
+	else if(flying)
+		inertial_drift()
+		make_floating(1)
+	//VOREStation Edit End
+	else if(!isspace(loc))
+		inertia_dir = 0
+		make_floating(0)
+
+/mob/living/proc/inertial_drift()
+	if(x > 1 && x < (world.maxx) && y > 1 && y < (world.maxy))
+		if(Process_Spacemove(1))
+			inertia_dir = 0
+			return
+		
+		var/locthen = loc
+		spawn(5)
+			if(!anchored && !pulledby && loc == locthen)
+				var/stepdir = inertia_dir ? inertia_dir : last_move
+				if(!stepdir)
+					return
+				var/turf/T = get_step(src, stepdir)
+				if(!T)
+					return
+				Move(T, stepdir, 5)
```

Comment:
Will this end up recursively calling inertial_drift() until it stops drifting?

Reply:
Well, I'm not sure I'd consider it 'recursively'. I guess there's a chain of things that happens that does come back to here until they're done drifting.
Move() into a turf that qualifies for inertial drift -> interial_drift() check to be sure, then 5 seconds later spawn a Move() -> check again to see if it still qualifies for drifting, repeat

S'the same as it was before, pretty much.